grails-app/controllers/com/unifina/controller/security/LoginController.groovy

```diff
@@ -54,8 +52,19 @@ class LoginController {
 
 		String view = 'auth'
 		String postUrl = "${request.contextPath}${config.apf.filterProcessesUrl}"
-		render view: view, model: [postUrl: postUrl,
-		                           rememberMeParameter: config.rememberMe.parameter]
+		def redirect = params.redirect
+		if (redirect != null) {
+			redirect = redirect.trim()
+			// Match https://www.streamr.com/ with required subdomain.
+			// Subdomain can be anything within [a-zA-Z0-9]
+			// Note required trailing slash!
+			if (!redirect.matches("^https://([a-zA-Z0-9]*)?\\.streamr\\.com/.*")) {
```

Comment:
We should also support `http://localhost:3333`, `http://127.0.0.1:3333` (or just `http://localhost:<ANYPORT>` and `http://127.0.0.1:<ANYPORT>` to future-proof any port changes)

Reply:
Done